# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/workers/ScriptLoader.cpp
# Commit: 6118d850a96d
# Full Hash: 6118d850a96dfa307d91cb73392c659ffb328335
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-11-18 15:46:32
# Regressor Bug: 1800496
# File Overlap Count: 1
# Description:
#   Bug 1800496 - Re-introduce channel tracking for cancellation; r=asuth
#   
#   This was removed as part of an attempt to simplify worker cancellaton. However, this channel cancellation is important, as without it the timing changes when we finish a request. Without it, we continue to load a script, even after a worker has been shut down.
#   
#   This tracking will help us reimplement cancellation.
# ==============================================================================

diff -r eafec005c1bf -r 6118d850a96d dom/workers/ScriptLoader.cpp
--- a/dom/workers/ScriptLoader.cpp	Fri Nov 18 10:02:26 2022 +0000
+++ b/dom/workers/ScriptLoader.cpp	Fri Nov 18 10:02:26 2022 +0000
@@ -981,6 +981,8 @@
     }
   }
 
+  loadContext->mChannel.swap(channel);
+
   return NS_OK;
 }
 
@@ -1056,6 +1058,7 @@
 
   WorkerLoadContext* loadContext = aRequest->GetWorkerLoadContext();
 
+  NS_ASSERTION(!loadContext->mChannel, "Should no longer have a channel!");
   NS_ASSERTION(aRequest->IsReadyToRun(), "Should be scheduled!");
 
   mRv.MightThrowJSException();